File: infer/tests/codetoanalyze/cpp/shared/methods/overloading.cpp
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

class A {
 public:
  int fun(int a, int b);
  int fun(int a, int b, int c);
};

int A::fun(int a, int b, int c) { return a + b + c; }

int A::fun(int a, int b) { return a - b; }

void test() {
  A* a_ptr;
  // calling methods
  a_ptr->fun(1, 2);
  a_ptr->fun(1, 2, 3);
}

File: infer/tests/codetoanalyze/java/sil/Overloading.java
//
// Copyright (c) Facebook, Inc. and its affiliates.
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.
//

class Overloading{
  int i;
  double f;

  public Overloading(int i, double f) {
    this.i = i;
    this.f = f;
  }

  public Overloading(int i) {
    this(i, 0.0);
  }

  public Overloading(double f) {
    this(0, f);
  }

  public Overloading() {
    this(0, 0);
  }

  public static Overloading make(int i) {
    return new Overloading(i);
  }

  public static Overloading make(double f) {
    return new Overloading(f);
  }

  public static Overloading make() {
    return new Overloading();
  }

}
